File: Nian_Karl/QL/src/org/uva/ql/ast/expression/binary/Equal.java
package org.uva.ql.ast.expression.binary;

import org.uva.ql.ast.builder.CodePosition;
import org.uva.ql.ast.expression.Expression;
import org.uva.ql.visitor.Visitor;

public class Equal extends Binary {

	public Equal(Expression left, Expression right, CodePosition pos) {
		super(left, right, pos);
	}

	@Override
	public <T> T accept(Visitor<T> visitor) {
		return visitor.visit(this);
	}
	
	@Override
	public String toString() {
		return this.left.toString() + " == " + this.right.toString();
	}
	
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expression/Binary/Equal.cs
using AST.Nodes.Interfaces;
using AST.Representation;

namespace AST.Nodes.Expression.Binary
{
    public class Equal : Binary, IExpression
    {
        public Equal(IExpression left, IExpression right, PositionInText position)
            :base(left, right, position)
        {}

        public override T Accept<T>(ASTVisitors.IVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return "==";
        }
    }
}
